common/environment/environment.cpp

```diff
@@ -1606,25 +1598,48 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
         addPathSepChar(fullDropZoneDir);
         const char * pdropzoneDir = fullDropZoneDir.str();
 
-        // Check target file path starts with this Drop zone path
-        // the drop zone paths can be nested (nothing forbids it) like
-        // dz1: /a/b/c/d
-        // dz2: /a/b/c
-        // dz3: /a/b
-        // check all and select the shortest
-
         if (strncmp(pdropzoneDir, targetFilePath, strlen(pdropzoneDir)) == 0)
         {
-            if (dropzonePathLen > strlen(pdropzoneDir))
+            dropZone = &zoneIt->query();
+
+            // For backward compatibility
+            SCMStringBuffer dropZoneMachineName;
+            // Returns dropzone '@server' value if it is exists
+            dropZone->getComputerName(dropZoneMachineName);
+
+            // check if it is local machine
+            if (strcmp(netaddress, dropZoneMachineName.str()) !=  0)
```

Comment:
This won't work/doesn't maintain backward compatibility. netaddress can't be compared to legacy @computer name...

Previously, netAddress was used to lookup a IConstMachineInfo with getMachineByAddress, then the name of that computer was matched against a DZ.
You still need to do that to maintain backward compatibility.

So _if_ getComputerName returns anything - then it is an old definition and you should perform same checks as before.